### DISCUSSION TITLE
Currently used InspectCode version doesn't support VS2022

### DISCUSSION THREAD

**Discussion by goodtrailer** [ppy/osu#17061] [Q&A] [answered]
VS2022 is required for .NET 6.0 but the currently used InspectCode version (2020.3.2) doesn't seem to support.
https://github.com/ppy/osu/blob/7aaacdda3dfaf91d781d9c5e40d0394a856c2d60/.config/dotnet-tools.json#L11-L16
According to [here](https://resharper-support.jetbrains.com/hc/en-us/articles/4404930135570-Support-for-Visual-Studio-2022), VS2022 is supported in InspectCode 2021.3 and later. Not sure if newer InspectCode versions work significantly different though (probably not).

**Comment by peppy** [comment]
I believe we have an old versions pinned to avoid a deadlock in CI runs, but I didn't think it was this old. @smoogipoo can you confirm the status of this one?

**Reply by smoogipoo** [reply]
Unfortunately it still deadlocks :( Progress has been slow on this from JetBrains' side, I'll poke them again about it.

More info here: https://youtrack.jetbrains.com/issue/RSRP-485500

**Comment by smoogipoo** [comment, marked as answer]
Newer R# versions will work fine for the time being. The `dotnet-tools.json` only affects the CLI app, and for the most part there's no difference between the two - a few edge cases that we've hit maybe once or twice.
